External/FEXCore/Source/Interface/HLE/Thunks/Thunks.cpp

```diff
@@ -56,6 +62,54 @@ namespace FEXCore {
           Thread->CTX->HandleCallback(Thread, (uintptr_t)callback);
         }
 
+        /**
+         * Instructs the Core to redirect calls to functions at the given
+         * address to another function. The original callee address is passed
+         * to the target function through an implicit argument stored in r11.
+         *
+         * The primary use case of this is ensuring that host function pointers
+         * returned from thunked APIs can safely be called by the guest.
+         */
+        static void LinkAddressToGuestFunction(void* argsv) {
+            struct args_t {
+                uintptr_t original_callee;
+                uintptr_t target_addr;     // Guest function to call when branching to replaced_addr
```

Comment:
```suggestion
                uintptr_t target_addr;     // Guest function to call when branching to original_callee
```

Reply:
I'd rather us rework the names in a way that the comments aren't needed

Reply:
I'm happy with the current balance between self-documenting code and descriptive docstrings. We're implementing highly unorthodox logic here, and a comment is the best tool to document the main idea here.